sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/ExtractKeyPhraseAsyncClient.java

```diff
@@ -53,38 +61,29 @@
     }
 
     /**
-     * Helper function for calling service with max overloaded parameters that a returns {@link TextAnalyticsPagedFlux}
-     * which is a paged flux that contains extracted key phrases.
+     * Helper function for calling service with max overloaded parameters that a returns {@link KeyPhrasesCollection}.
      *
      * @param document A document.
      * @param language The language code.
      *
-     * @return The {@link TextAnalyticsPagedFlux} of extracted key phrases strings.
+     * @return The {@link Mono} of {@link KeyPhrasesCollection} extracted key phrases strings.
      */
-    TextAnalyticsPagedFlux<String> extractKeyPhrasesSingleText(String document, String language) {
-        Objects.requireNonNull(document, "'document' cannot be null.");
-        return new TextAnalyticsPagedFlux<>(() ->
-            (continuationToken, pageSize) -> extractKeyPhrases(
-                Collections.singletonList(new TextDocumentInput("0", document, language)), null).byPage()
-                .map(resOfResult -> {
-                    Iterator<ExtractKeyPhraseResult> iterator = resOfResult.getValue().iterator();
-                    // Collection will never empty
-                    if (!iterator.hasNext()) {
-                        throw logger.logExceptionAsError(new IllegalStateException(
-                            "An empty collection returned which is an unexpected error."));
-                    }
-
-                    final ExtractKeyPhraseResult keyPhraseResult = iterator.next();
-                    if (keyPhraseResult.isError()) {
-                        throw logger.logExceptionAsError(
-                            Transforms.toTextAnalyticsException(keyPhraseResult.getError()));
-                    }
-
-                    return new TextAnalyticsPagedResponse<>(
-                        resOfResult.getRequest(), resOfResult.getStatusCode(), resOfResult.getHeaders(),
-                        keyPhraseResult.getKeyPhrases().stream().collect(Collectors.toList()),
-                        null, resOfResult.getModelVersion(), resOfResult.getStatistics());
-                }));
+    Mono<KeyPhrasesCollection> extractKeyPhrasesSingleText(String document, String language) {
+        try {
+            Objects.requireNonNull(document, "'document' cannot be null.");
+            final TextDocumentInput textDocumentInput = new TextDocumentInput("0", document);
+            textDocumentInput.setLanguage(language);
+            return extractKeyPhrases(Collections.singletonList(textDocumentInput), null)
+                    .map(keyPhraseResult -> {
+                        if (keyPhraseResult.isError()) {
```

Comment:
What's the expect behavior here?
1. Stop doing other maps if one event gets throwable
1. Keep doing other maps if one event gets throwable

Reply:
1. Stop doing other maps if one event gets throwable

Reply:
in this case, since it is single-input API, which take one input document. But internally, we call the batch API. Which could have the result is error document. If the result is error, for single-input API, we throw the TextAnalyticsException but not return error.

Reply:
talked offline, no longer an issue.